test/Microsoft.Health.Dicom.Web.Tests.E2E/Common/DicomInstancesManager.cs

```diff
@@ -89,6 +89,27 @@ public async Task<DicomWebResponse<DicomDataset>> StoreAsync(Stream stream, stri
         return await _dicomWebClient.StoreAsync(stream, studyInstanceUid, partitionName, cancellationToken);
     }
 
+    public async Task StoreIfNotExistsAsync(DicomFile dicomFile, bool doNotDelete = false, string partitionName = default, CancellationToken cancellationToken = default)
+    {
+        EnsureArg.IsNotNull(dicomFile, nameof(dicomFile));
+
+        var dicomInstanceId = DicomInstanceId.FromDicomFile(dicomFile, partitionName);
+
+        try
+        {
+            await _dicomWebClient.RetrieveInstanceAsync(dicomInstanceId.StudyInstanceUid, dicomInstanceId.SeriesInstanceUid, dicomInstanceId.SopInstanceUid, cancellationToken: cancellationToken);
+        }
+        catch (DicomWebException)
+        {
+            await _dicomWebClient.StoreAsync(dicomFile, dicomInstanceId.StudyInstanceUid, partitionName, cancellationToken);
+
+            if (!doNotDelete)
+            {
+                _instanceIds.Add(dicomInstanceId);
```

Comment:
Do we need this amount of flexibility if we always store the instance, if necessary, at the start of the test?

Reply:
i dont want to make this so generic, just want to scope to this test only.

Reply:
I added this flag, so that future implementation will not accidently not delete